src/communication/stream_transfer_receiver.hpp

```diff
@@ -54,7 +55,8 @@ namespace cubstream
 
       friend class transfer_receiver_task;
 
-      cubcomm::channel m_channel;
+      cubcomm::channel m_repl_channel;
+      cubcomm::channel m_ctrl_channel;
```

Comment:
Stream transfer is a generic interface that could be used in other than replication; control channel seems more tightly coupled to HA replication logic.

What do you think about adding the control channel to cubreplication::slave_node?